src/Features/Core/Portable/Completion/Providers/Scripting/AbstractDirectivePathCompletionProvider.cs

```diff
@@ -60,8 +65,33 @@ public sealed override async Task ProvideCompletionsAsync(CompletionContext cont
             }
         }
 
-        public override bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
-            => true;
+        public sealed override bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
+        {
+            var lineStart = text.FindContainingLineStart(caretPosition);
+
+            // check if the line starts with {whitespace}#{whitespace}{DirectiveName}{whitespace}"
+
+            var poundIndex = text.IndexOfNonWhiteSpace(lineStart, caretPosition - lineStart);
```

Comment:
`GetFirstNonWhitespacePosition` calls `line.ToString()`. Is that better then indexing into SourceText?

Reply:
The new extension method is more primitive than `GetFirstNonWhitespacePosition` since `TextLine` does not have an indexer, so `GetFirstNonWhitespacePosition` would call the new one.
There are multiple places and methods that can be optimized in a follow-up. Filed issue https://github.com/dotnet/roslyn/issues/50706.